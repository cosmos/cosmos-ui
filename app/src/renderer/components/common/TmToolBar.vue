<template lang="pug">
  tm-tool-bar
    slot
    a.back(@click="$router.go(-1)" :disabled="user.history.length === 0" v-tooltip.bottom="'Back'")
      i.material-icons arrow_back
    a.help(@click="enableModalHelp" v-tooltip.bottom="'Help'")
      i.material-icons help_outline
    a.sign-out(@click="signOut" v-tooltip.bottom.end="'Sign Out'")
      i.material-icons exit_to_app
</template>

<script>
import { TmToolBar } from "@tendermint/ui"
import { mapGetters } from "vuex"
export default {
  // the name needs to be different from TmToolBar (tm-tool-bar) or else recursive rendering takes place
  name: "vm-tool-bar",
  components: {
    TmToolBar
  },
  methods: {
    enableModalHelp() {
      this.$store.commit("setModalHelp", true)
    },
    signOut() {
      this.$store.dispatch("signOut")
    }
  },
  computed: {
    ...mapGetters(["user"])
  }
}
</script>
