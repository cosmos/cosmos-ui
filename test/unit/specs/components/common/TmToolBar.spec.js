import setup from "../../../helpers/vuex-setup"
import TmToolBar from "common/TmToolBar"

describe("TmToolBar", () => {
  let wrapper, store, router
  let { mount } = setup()

  beforeEach(() => {
    let instance = mount(TmToolBar)
    router = instance.router
    wrapper = instance.wrapper
    store = instance.store
    wrapper.update()
  })

  it("has the expected html structure", () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("sets the helper modal", () => {
    wrapper.vm.enableModalHelp()
    wrapper.update()
    expect(store.state.config.modals.help.active).toBe(true)
  })

  it("call dispatch to sign the user out", () => {
    wrapper.vm.signOut()
    wrapper.update()
    expect(store.dispatch).toHaveBeenCalledWith("signOut")
  })

  it("goes back correctly and updates the state", () => {
    expect(router.currentRoute.fullPath).toBe("/")

    router.push("/staking")
    expect(store.state.user.history.length).toBe(1)
    expect(router.currentRoute.fullPath).toBe("/staking")

    wrapper.vm.back()
    expect(store.state.user.history.length).toBe(0)
    expect(router.currentRoute.fullPath).toBe("/")

    wrapper.vm.back()
    expect(store.state.user.history.length).toBe(0)
    expect(router.currentRoute.fullPath).toBe("/")
  })
})
