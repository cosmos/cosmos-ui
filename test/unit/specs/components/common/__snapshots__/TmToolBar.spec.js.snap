// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmToolBar has the expected html structure 1`] = `"<div class=\\"tm-tool-bar\\"><div class=\\"tm-tool-bar-container\\"><div class=\\"main\\"><a disabled=\\"disabled\\" class=\\"back\\"><i class=\\"material-icons\\">arrow_back</i></a><a class=\\"help\\"><i class=\\"material-icons\\">help_outline</i></a><a class=\\"sign-out\\"><i class=\\"material-icons\\">exit_to_app</i></a></div></div></div>"`;
